pkg/lockfile/parse-npm-lock.go

```diff
@@ -57,18 +59,43 @@ func pkgDetailsMapToSlice(m map[string]PackageDetails) []PackageDetails {
 	return details
 }
 
-func mergePkgDetailsMap(m1 map[string]PackageDetails, m2 map[string]PackageDetails) map[string]PackageDetails {
-	details := map[string]PackageDetails{}
-
-	for name, detail := range m1 {
-		details[name] = detail
+type npmPackageDetailsMap map[string]PackageDetails
+
+// mergeNpmDepsGroups handles merging the dependency groups of packages within the
+// NPM ecosystem, since they can appear multiple times in the same dependency tree
+//
+// the merge happens almost as you'd expect, except that if either given packages
+// belong to no groups, then that is the result since it indicates the package
+// is implicitly a production dependency.
+func mergeNpmDepsGroups(a, b PackageDetails) []string {
```

Comment:
it seems we only need `DepGroups` so can the func be
```
func mergeNpmDepGroups(a, b []string) []string
```
?

Reply:
It could but that'd mean every caller of this function would have to explicitly access `DepGroups` themselves - so it's cleaner for us to take `PackageDetails` and do that accessing ourselves.